Diakonie 503 deletion issue with agency

Leandro13Silva13 opened the pull request.

Leandro13Silva13 committed:
DIAKONIE-503

change userService api call to the correct one

Leandro13Silva13 committed:
DIAKONIE-503

update actions/download-artifact

vi-sudo commented:
Task linked: [OB-20157 [Diakonie] Task: check deletion issue with agency ](https://app.clickup.com/t/6656196/OB-20157)

patric-dosch-vi commented on src/main/java/de/caritas/cob/agencyservice/api/admin/service/UserAdminService.java:
```diff
@@ -88,8 +88,6 @@ public List<ConsultantAdminResponseDTO> getConsultantsOfAgency(Long agencyId, in
-    return controllerApi
-        .getConsultants(currentPage, perPage, consultantFilter, sortBy)
-        .getEmbedded();
+    return controllerApi.getAgencyConsultants(String.valueOf(agencyId)).getEmbedded();
```
hmm is this method only used for the usecase of deletion?
Because you are changing the logic here for sure. If it's only used for deletion, then we should also remove currentPage, perPage and filter... also in the signature of the method.

Currently, what you are changing here is really far from clean code :/

Leandro13Silva13 replied:
This is only used to get all the users that are in a agency, so this is actually the API that should have always been called.
I also agree that this is not a clean code, I will remove everything that is not needed anymore for the new API call.

Leandro13Silva13 committed:
DIAKONIE-503

clean code, by removing unused variables and parameters from UserAdminService.getConsultantsOfAgency

Leandro13Silva13 committed:
DIAKONIE-503

fix failing tests prior to my changes

tkuzynow reviewed (changes requested):
please fix the build

Leandro13Silva13 committed:
DIAKONIE-503

fix build error and failing tests prior to my changes
